packages/pass-style/src/copyBytes.js

```diff
@@ -0,0 +1,82 @@
+/// <reference types="ses"/>
+
+import { assertChecker } from './passStyle-helpers.js';
+
+/** @typedef {import('./types.js').CopyBytes} CopyBytes */
+
+const { Fail } = assert;
+const { setPrototypeOf } = Object;
+const { apply } = Reflect;
+
+/**
+ * @type {WeakSet<CopyBytes>}
+ */
+const genuineCopyBytes = new WeakSet();
+
+const slice = ArrayBuffer.prototype.slice;
+const sliceOf = (buffer, start, end) => apply(slice, buffer, [start, end]);
+
+/**
+ * A CopyBytes is much like an ArrayBuffer, but immutable.
+ * It cannot be used as an ArrayBuffer argument when a genuine ArrayBuffer is
+ * needed. But a `copyBytes.slice()` is a genuine ArrayBuffer, initially with
+ * a copy of the copyByte's data.
+ *
+ * On platforms that support freezing ArrayBuffer, like perhaps a future XS,
+ * (TODO) the intention is that `copyBytes` could hold on to a single frozen
+ * one and return it for every call to `arrayBuffer.slice`, rather than making
+ * a fresh copy each time.
```

Comment:
> whether there should be a JS API to create an immutable ArrayBuffer from another one

We've proposed [snapshot](https://github.com/tc39/proposal-readonly-collections#snapshotdivergereadonlyview-methods-for-all-collections) for that, yes?

odd... https://github.com/tc39/proposal-readonly-collections#status says stage 1 but I don't see it inhttps://github.com/tc39/proposals

Reply:
That proposal seem targeted at collections, not ArrayBuffer. There is however a related proposal for ArrayBuffer: https://github.com/tc39/proposal-limited-arraybuffer

Reply:
Arrays aren't collections? hm.

Reply:
ArrayBuffers aren't arrays.

Reply:
We're no longer trying to base this on https://github.com/tc39/proposal-readonly-collections#snapshotdivergereadonlyview-methods-for-all-collections but rather a `transferToImmutable` to be proposed, as shimmed at #2311 . Resolving.

Reply:
@mhofman , thanks for the pointer to https://github.com/tc39/proposal-limited-arraybuffer !

@Jack-Works , what do you think of `transferToImmutable` as an alternate way to achieve these goals? (See shim at #2311 of suggestion by @phoddie at https://github.com/endojs/endo/pull/2309#issuecomment-2155513240 .) Since proposal-limited-arraybuffer is at stage 1, if you like this alternate route to these goals, I'd consider it within stage 1 to shift that proposal in this direction. If you agree, should we work together to revise proposal-limited-arraybuffer in this way? I'd certainly be happy to co-champion.

Reply:
Limited ArrayBuffer has stated goals that go beyond a simple view on an immutable ArrayBuffer. There are likely orthogonal aspects.